### DISCUSSION TITLE
How can I enable random seed in GUI?

### DISCUSSION THREAD

**Discussion by Sefrautic** [JonathanFly/bark#52] [Q&A]
And also if this possilble, can I see somewhere what seed was used for the particular generation?

**Comment by JonathanFly** [comment]
The seed feature is on the the dev branch.

If you type 'git checkout dev' from the same place you type 'python bark_whatever.py'  you can switch to the dev branch.

Oh wait there is a seed even on the non dev. It's on the far right next to where you set the folder. I'm not positive it works in the non-dev version though!

You can't look at the seed for past generations. It would be possible to always record the seed but you do take a pretty big speed hit when you turn it on. Though if you turn it halfway on I think you get really close audio, without the speed hit, which might be worth trying.

**Reply by Sefrautic** [reply]
> Oh wait there is a seed even on the non dev. It's on the far right next to where you set the folder. I'm not positive it works in the non-dev version though!

Yeah I noticed that, but it's confusing. I mean, I write some random numbers and the output doesn't change (probably it doesn't work, as you said). And I don't understand what I should write to enable random seed every time.

Maybe I'll try the dev branch, thanks.

Also, in the GUI there is no options to enable infinite length. I get 14 sec max. Maybe I should enter some arguments in the arguments field?

**Comment by JonathanFly** [comment]
What should happen is if you put in too much text, the text gets split into two clips and then recombined. If that didn't happen did you adjust the text settings? It should by default split it kind of reasonably if you type in a full page of text, for example.

Basically it should always be in infinite length mode, unless you specifically turn that off.

The seed is only used to PREVENT the output from changing. It's so you can make the exact same voice, perfect match.
